src/ulid.cr

```diff
@@ -19,6 +19,57 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # Incorrect length
+    return false unless ulid.size == TIME_LEN + RANDOM_LEN
+    # Invalid chars
+    return false unless ulid.upcase.chars.all? &.in?(ENCODING)
+    
+    return true
+  end
+
+  # Validate a string is a ULID, raises errors for invalid strings
+  #
+  # ```
+  # ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => nil
+  # ```
+   def valid!(ulid : String) : Nil
+    raise IncorrectLength.new("ULID should be 26 characters") unless ulid.size == TIME_LEN + RANDOM_LEN
+    raise InvalidChars.new("Invalid characters found in string. Should only contain #{ENCODING}") unless ulid.upcase.chars.all? &.in?(ENCODING)
```

Comment:
Should 
`ulid.size == TIME_LEN + RANDOM_LEN `
and 
`ulid.upcase.chars.all? &.in?(ENCODING) `
be refactored into their own methods?

Reply:
I'd keep them together. There's no additional information outside of if it's a valid ULID or not.

On a similar note, I would avoid raising exceptions here. As a user of the lib, the main interest is just checking validity, which can be handled most simply with a plain, boring `ULID.valid?(String) : Bool`. There is no possible exceptional circumstance that can occur based on what is is encoded into the type `String -> Bool`.